molecule/api.py

```diff
@@ -1,26 +1,50 @@
-import pkg_resources
+import pluggy
+from importlib import import_module
 from molecule import logger
-
+import traceback
 
 LOG = logger.get_logger(__name__)
-
 try:
     from functools import lru_cache
 except ImportError:
     from backports.functools_lru_cache import lru_cache
 
 
-@lru_cache()
-def molecule_drivers(as_dict=False):
+CORE_DRIVERS = [
+    "delegated",
+    "digitalocean",
+    "docker",
+    "ec2",
+    "gce",
+    "hetznercloud",
+    "linode",
+    "lxc",
+    "lxd",
+    "openstack",
+    "podman",
+    "vagrant",
+]
 
+
+@lru_cache()
+def molecule_drivers(as_dict=False, config=None):
     plugins = {}
-    for entry_point in pkg_resources.iter_entry_points('molecule_driver'):
+    pm = pluggy.PluginManager("molecule_driver")
+    for driver in CORE_DRIVERS:
+        pm.register(import_module("molecule.driver.%s" % driver), name=driver)
+    try:
+        pm.load_setuptools_entrypoints("molecule_driver")
+    except Exception:
+        # These are not fatal because a broken driver should not make the entire
+        # tool unusable.
+        LOG.error("Failed to load driver entry points %s", traceback.format_exc())
+    for p in pm.get_plugins():
         try:
-            plugins[entry_point.name] = entry_point.load()
+            plugins[pm.get_name(p)] = p.load(config)
         except Exception as e:
-            LOG.error("Failed to load %s driver: %s", entry_point.name, str(e))
+            LOG.error("Failed to load %s driver: %s", pm.get_name(p), str(e))
 
     if as_dict:
         return plugins
     else:
-        return list(plugins.keys())
+        return sorted(list(plugins.keys()))
```

Comment:
why does this list need to be sorted? Not that the action has much impact, just a question as to why this is being done for an internal method?

Reply:
if we do need/want to sort the return we could simply run the following

``` python
sorted(plugins.keys())
```

Reply:
As far as I understand, the `molecule drivers` command will list them? So, this list ends up being shown to the user. I haven't checked the code though. If that is right, sorting makes sense.

Reply:
As far as I understand, the `molecule drivers` command will list them? So, this list ends up being shown to the user. I haven't checked the code though. If that is right, sorting makes sense.

Reply:
If that's the case, then :+1: